mountpoint-s3/tests/fuse_tests/fork_test.rs

```diff
@@ -705,3 +922,42 @@ fn unmount_and_check_log(mut process: Child, mount_path: &Path, expected_log_lin
     }
     panic!("can not find a matching line in log: [{log}]");
 }
+
+/// Create a CLI config file containing a profile that source its credentials from another given profile.
+fn create_cli_config_file(
+    profile_name: &str,
+    source_profile: &str,
+    role_arn: &str,
+    region: &str,
+) -> io::Result<NamedTempFile> {
+    let mut config_file = NamedTempFile::new()?;
+
+    writeln!(config_file, "[profile {}]", profile_name).unwrap();
+    writeln!(config_file, "source_profile = {}", source_profile).unwrap();
+    writeln!(config_file, "role_arn = {}", role_arn).unwrap();
+    writeln!(config_file, "region = {}", region).unwrap();
+    Ok(config_file)
+}
+
+/// Create a shared credentials file from static credentials in environment variables.
+/// Return a `NamedTempFile` if `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are set, otherwise None.
+fn create_shared_credentials_file_if_possible(profile_name: &str) -> io::Result<Option<NamedTempFile>> {
+    match get_credentials_from_env() {
+        Some(credentials) => {
+            let mut credentials_file = NamedTempFile::new()?;
+            writeln!(credentials_file, "[{}]", profile_name).unwrap();
+            writeln!(credentials_file, "aws_access_key_id={}", credentials.access_key_id()).unwrap();
+            writeln!(
+                credentials_file,
+                "aws_secret_access_key={}",
+                credentials.secret_access_key()
+            )
+            .unwrap();
+            if let Some(session_token) = credentials.session_token() {
+                writeln!(credentials_file, "aws_session_token={session_token}").unwrap();
+            }
+            Ok(Some(credentials_file))
+        }
+        None => Ok(None),
+    }
+}
```

Comment:
To keep things simple, I'd suggest we move to an approach where we always create the config file entry for the source profile (so its consistent between local and CI).

Would it be possible to also use the default chain like the `auth` tests. I know we'll need to block on the future / async fn which we didn't do in this module yet.

https://github.com/awslabs/mountpoint-s3/blob/8c3e18e80a2fa1ee53eaf5cb5c286d003609683c/mountpoint-s3-client/tests/auth.rs#L43-L51